controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/UpdateStreamTask.java

```diff
@@ -51,34 +52,41 @@ public UpdateStreamTask(final StreamMetadataTasks streamMetadataTasks,
         String scope = request.getScope();
         String stream = request.getStream();
 
-        return streamMetadataStore.getConfigurationRecord(scope, stream, true, context, executor)
-                .thenCompose(configProperty -> {
-                    if (!configProperty.isUpdating()) {
-                        // if the state is updating but the configuration record is not updating, we should reset the state to ACTIVE.
-                        return streamMetadataStore.resetStateConditionally(scope, stream, State.UPDATING, context, executor)
-                                .thenRun(() -> {
+        return streamMetadataStore.getVersionedState(scope, stream, context, executor)
+                .thenCompose(versionedState -> streamMetadataStore.getConfigurationRecord(scope, stream, context, executor)
+                        .thenCompose(versionedMetadata -> {
+                            if (!versionedMetadata.getObject().isUpdating()) {
```

Comment:
I don't understand this logic:

```
if (!versionedMetadata.getObject().isUpdating()) {
           if (versionedState.getObject().equals(State.UPDATING)) {

```

Why is it expected that the the metadata object is updating? I thought from the PR description that the versioned state had to be checked instead and had to be something like `ACTIVE`.

Reply:
refer to the response here: https://github.com/pravega/pravega/pull/2971/files#r228244936

in a rerun if we find State = State.Updating without corresponding metadata indicating an ongoing/outstanding processing, this can only happen if its a rerun of a job that had completed the work but crashed before resetting the state. 
In such cases we reset the state and exit the job.

Reply:
Hopefully I'm reading this right, but you seem to be saying:

```
if (state is updating) {
   if(metadata is not updating) {
       reset state to active
   }
}
```

while the logic in the code is the opposite:

```
if (!versionedMetadata.getObject().isUpdating()) {
         if (versionedState.getObject().equals(State.UPDATING)) {
                                    return Futures.toVoid(streamMetadataStore.updateVersionedState(scope, stream, State.ACTIVE, versionedState, context, executor));
         }
}
```